### ISSUE TITLE
Error when running part1

### ISSUE THREAD

**Issue by xiekunwhy** [AO33/HiC_Genome_Assembler#1]
Hi,

I am trying to run your pipeline, but a got following errors, do you know why?
Time to convert matrix 13.637509822845459
Time to condensed matrix 0.884394645690918
Time to cluster 5.115649700164795
Traceback (most recent call last):
  File "/Bio/User/kxie/pipeline/genome/anchor/apps/HIC_ASSEMBLER/run_hicAssembler.py", line 212, in <module>
    part1.runPipeline(varDict["hicProBedFile"],varDict["hicProBiasFile"],varDict["hicProMatrixFile"],varDict["hicProScaffSizeFile"],
  File "/Bio/User/kxie/pipeline/genome/anchor/apps/HIC_ASSEMBLER/scaffoldToChromosomes.py", line 596, in runPipeline
    plotModule.plotContactMap(adjMat,resolution=resolution,highlightChroms=False,showPlot=False,savePlot=avgClusterPlot)
  File "/Bio/User/kxie/pipeline/genome/anchor/apps/HIC_ASSEMBLER/plotContactMaps.py", line 25, in plotContactMap
    vmin=numpy.percentile(adjMat,lP),
  File "<__array_function__ internals>", line 180, in percentile
  File "/Bio/User/kxie/software/anaconda3/envs/hicassembler/lib/python3.10/site-packages/numpy/lib/function_base.py", line 4166, in percentile
    return _quantile_unchecked(
  File "/Bio/User/kxie/software/anaconda3/envs/hicassembler/lib/python3.10/site-packages/numpy/lib/function_base.py", line 4424, in _quantile_unchecked
    r, k = _ureduce(a,
  File "/Bio/User/kxie/software/anaconda3/envs/hicassembler/lib/python3.10/site-packages/numpy/lib/function_base.py", line 3725, in _ureduce
    r = func(a, **kwargs)
  File "/Bio/User/kxie/software/anaconda3/envs/hicassembler/lib/python3.10/site-packages/numpy/lib/function_base.py", line 4593, in _quantile_ureduce_func
    result = _quantile(arr,
  File "/Bio/User/kxie/software/anaconda3/envs/hicassembler/lib/python3.10/site-packages/numpy/lib/function_base.py", line 4710, in _quantile
    result = _lerp(previous,
  File "/Bio/User/kxie/software/anaconda3/envs/hicassembler/lib/python3.10/site-packages/numpy/lib/function_base.py", line 4529, in _lerp
    lerp_interpolation = asanyarray(add(a, diff_b_a * t, out=out))
  File "/Bio/User/kxie/software/anaconda3/envs/hicassembler/lib/python3.10/site-packages/numpy/matrixlib/defmatrix.py", line 218, in __mul__
    return N.dot(self, asmatrix(other))
  File "<__array_function__ internals>", line 180, in dot
ValueError: shapes (1,115648516) and (1,1) not aligned: 115648516 (dim 1) != 1 (dim 0)

Best,
Kun

**Comment by AO33** [comment]
Hmm, my best guess at the moment is that line 595 in scaffoldToChromosomes.py needs to be changed to numpy.asmatrix instead of scipy.asmatrix.

Also note, I am about to update with a new version that allows for a different option instead of the hmm to cut the avg clustering dendrogram that should hopefully be more stable. I recommend using 500Kb if you have a large genome size to speed things up (or at least until the program is running without error)

**Comment by xiekunwhy** [comment]
Thank you for your reply, I use 100Kb for a 600Mb genome.

part1 worked well when I comment out all plotModule relate lines in scaffoldToChromosomes.py.

And as you said, part2 is very slow, still running now.

I am waiting for the new updated verison.

Best,
Kun

**Comment by AO33** [comment]
No problem :)
I just updated with the newest version. Simply redownload the directory with the new code and the updated config file will be included. Should run the same as before... I have added a print statement in the plotContactMap function to check the shape of your input matrix which might help debug / determine if it is an issue with the input HiCpro files. 

Note that psig, hyperGeom, hmm, and lookAhead are new options that were added.

I also recommend taking a look at the jupyter notebook version if you are able. Might make it easier to debug.

**Comment by xiekunwhy** [comment]
Thank you, I will try it.

By the way, I found that numba require elder python version than xarray numpy and scipy, so numba should be install first, then others.

Best,
Kun

**Comment by AO33** [comment]
Great. Thanks for reporting that. I will update the documentation soon to reflect this

**Comment by AO33** [comment]
I was able to reproduce your error on my end by reordering the installation of the various packages. Looking into how to resolve this

**Comment by AO33** [comment]
I was able to fix the bug you encountered. The numpy percentile function prefers an array type rather than a matrix type for certain configurations of numba + numpy apparently. 

The plotContactMaps.py module has been updated accordingly to ensure that data is converted to an array rather than matrix.

The heatmap visualization is the easiest way to tell if the program is working properly or not (in my opinion) so hopefully this will solve the issue in it's entirety. Cheers
